airflow/upgrade/rules/send_grid_moved.py

```diff
@@ -0,0 +1,38 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from __future__ import absolute_import
+
+from airflow.configuration import conf
+from airflow.upgrade.rules.base_rule import BaseRule
+
+
+class SendGridEmailerMovedRule(BaseRule):
+    title = "SendGrid email uses old airflow.contrib module"
+
+    description = """
+Removes the need for SendGrid email code to be in contrib package. The SendGrid email function \
+has been moved to airflow.providers.
+    """
+
+    def check(self):
+        email_conf = conf.get(section="email", key="email_backend")
+        if email_conf.startswith("airflow.contrib"):
```

Comment:
I think it would be better to compare it to old sendgrid path

Reply:
Hi @turbaszek! Do you mean `airflow.contrib.utils.sendgrid`? I figured if we pick up config with `airflow.contrib` it will be the older package as opposed to `airflow.providers`.

Reply:
> Do you mean airflow.contrib.utils.sendgrid

Yes, this gives better precision than just `airflow.contrib` and this rule focuses on sendgrind not whole contrib 😉

Reply:
Sure, makes sense 👌🏼